src/screens/NewSubmission.tsx

```diff
@@ -43,25 +43,28 @@ export default function NewSubmissionScreen({ navigation }: DrawerScreenProps<Dr
     resolver: zodResolver(schema),
   });
 
-  const onSubmit: SubmitHandler<NewSubmissionFormData> = (data) => {
-    setIsLoading(true);
-    axiosInstance
-      .post('/submissions', data)
-      .then(response => {
-        setApiError(null);
-        navigation.navigate('HomeStack', {
-          screen: 'Home',
-          params: { newSubmissionAdded: true },
-        });
-        reset();
-      })
-      .catch(error => {
-        console.log(error.response);
-        const message = getErrorMessage(error);
-        setApiError(message);
-      }).finally(() => {
-        setIsLoading(false);
+
+  const createSubmission = (data: NewSubmissionFormData) => axiosInstance.post('/submissions', data);
+
+  const { mutate: createSubmissionMutation, isLoading } = useMutation({
+    mutationFn: createSubmission,
+    onSuccess: () => {
+      setApiError(null);
+      navigation.navigate('HomeStack', {
+        screen: 'Home',
+        params: { newSubmissionAdded: true },
       });
+      reset();
```

Comment:
if you're navigating you no longer need to reset the form right?

Reply:
Even if I navigate, if I don't reset it, for some reason when I go back the inputs still have what I wrote before